Geometry/HGCalGeometry/src/HGCalGeometry.cc

```diff
@@ -300,19 +300,29 @@ HGCalGeometry::CornersVec HGCalGeometry::getCorners(const DetId& detid) const {
     if ((mode_ == HGCalGeometryMode::Hexagon) ||
 	(mode_ == HGCalGeometryMode::HexagonFull)) {
       xy = m_topology.dddConstants().locateCellHex(id.iCell1,id.iSec1,true);
+      float dx = m_cellVec[cellIndex].param()[FlatHexagon::k_r];
+      float dy = k_half*m_cellVec[cellIndex].param()[FlatHexagon::k_R];
+      float dz = m_cellVec[cellIndex].param()[FlatHexagon::k_dZ];
+      static const int signx[] = {0,-1,-1,0,1,1,0,-1,-1,0,1,1};
+      static const int signy[] = {-2,-1,1,2,1,-1,-2,-1,1,2,1,-1};
+      static const int signz[] = {-1,-1,-1,-1,-1,-1,1,1,1,1,1,1};
+      for (unsigned int i = 0; i < ncorner; ++i) {
+	const HepGeom::Point3D<float> lcoord(xy.first+signx[i]*dx,xy.second+signy[i]*dy,signz[i]*dz);
+	co[i] = m_cellVec[cellIndex].getPosition(lcoord);
+      }
     } else {
       xy = m_topology.dddConstants().locateCell(id.iLay,id.iSec1,id.iSec2,
 						id.iCell1,id.iCell2,true,false);
-    }
-    float dx = m_cellVec[cellIndex].param()[FlatHexagon::k_r];
-    float dy = k_half*m_cellVec[cellIndex].param()[FlatHexagon::k_R];
-    float dz = m_cellVec[cellIndex].param()[FlatHexagon::k_dZ];
-    static const int signx[] = {0,-1,-1,0,1,1,0,-1,-1,0,1,1};
-    static const int signy[] = {-2,-1,1,2,1,-1,-2,-1,1,2,1,-1};
-    static const int signz[] = {-1,-1,-1,-1,-1,-1,1,1,1,1,1,1};
-    for (unsigned int i = 0; i < ncorner; ++i) {
-      const HepGeom::Point3D<float> lcoord(xy.first+signx[i]*dx,xy.second+signy[i]*dy,signz[i]*dz);
-      co[i] = m_cellVec[cellIndex].getPosition(lcoord);
+      float dx = k_fac2*m_cellVec[cellIndex].param()[FlatHexagon::k_r];
+      float dy = k_fac1*m_cellVec[cellIndex].param()[FlatHexagon::k_R];
+      float dz =-id.zSide*m_cellVec[cellIndex].param()[FlatHexagon::k_dZ];
```

Comment:
It looks like dx, dy, and dz are not changed after initialization. It would be good to make them `const`. Same for lines 303-305.

Reply:
It is an incorrect judgement. Though dx, dy, dz may be the same for a large number of cells, they will be different for all cells. We have so called fine/coarse cells and also with different depletion thickness.  Please sign this PR

Reply:
I was merely suggesting changing, for example, line 316 from:
`float dx = k_fac2*m_cellVec[cellIndex].param()[FlatHexagon::k_r];`
to
`const float dx = k_fac2*m_cellVec[cellIndex].param()[FlatHexagon::k_r];`
but not moving or otherwise changing the definition of `dx`. `dx` from line 316 is read once and then goes out of scope. It is not changed, so it is a constant.
Our CMS coding guidelines say that all variables should be declared `const` unless they need to be changed ([Rule 7.5](https://github.com/cms-sw/cms-sw.github.io/blob/a27560c6ecb214893b5ef31fc6dd935fb7c2288c/cms_coding_rules.md#7----design-and-coding-guidelines-1)). The coding guidelines are still under review right now, so it would be good for all developers to take a look at them to see if any changes are needed.
It's a very minor issue. The `const` declaration would just help someone reading the code understand that `dx`, `dy`, and `dz` don't change after being initialized.